### DISCUSSION TITLE
How to apply inverse_transform on PAA?

### DISCUSSION THREAD

**Discussion by buffalocu** [alan-turing-institute/sktime#3911] [Q&A]
I'd like to reverse_transform a PAA. How can I do it? 
The PAA code is below: 

```
from sktime.transformations.panel.dictionary_based._paa import PAA
from sktime.utils._testing.panel import _make_nested_from_array
X = _make_nested_from_array(np.random.rand(10), n_instances=10, n_columns=1)
for i in range(0,10):
    X.col0[i] = pd.Series(np.random.rand(10))
XPAA = PAA(num_intervals = 3).fit(X).transform(X)

```
How to get XPAA.inverse_transform? 

Additinally, when PAA transform applies to a panel data, will the piecewise transfromation will be the same for all samples in the panel? For  example, there are 10 timestep samples for each instance and 100 intances in the panel. Will the 100 instance be piecewise for the same timestep, such as piece one all for time step 1-3,piece two  all for time step 4-7, and piece tree all for 8-10? Or each instance is piecewise seperately based on its own charactors?  scuh as instance 1 piece one from time step 1-2, piece two from time step 2-8, piece three from time step 9-10, while other instance could result in piece one from 1-3, piece two from 4-7, piece three from 8-10?

**Comment by fkiraly** [comment]
The `PAA` transformer currently does not have an inverse transform.

In general, this can have two reasons - "does not make methodological sense", or "has not been implemented". I have not looked into the paper so cannot say which of these it is.

* to check whether a transformer has inverse transform, do `my_trafo.get_tag("capability:inverse_transform")` - you can find more tags with explanations by using `registry.all_tags`
* for transformers that *do* have an `inverse_transform`, simply call `my_trafo.inverse_transform(X)` - as stated `PAA` does not support that.
* to implement an inverse transform for a transformer, fill in the `_inverse_transform` method using the transformer extension template, here is the [developer documentation](https://www.sktime.org/en/stable/developer_guide/add_estimators.html) with a link to a tutorial video if you are interested

**Reply by buffalocu** [reply]
Thank you, Fkiraly!

**Comment by fkiraly** [comment]
> Additinally, when PAA transform applies to a panel data, will the piecewise transfromation will be the same for all samples in the panel?

Quickly looked at the code - it looks like it is the same, as there is no randomisation in the feature extraction and the algorithm is the same for all instances, see `_perform_paa_along_dim`
